packages/server-wallet/src/protocols/state.ts

```diff
@@ -0,0 +1,46 @@
+import {SignedStateWithHash} from '@statechannels/wallet-core';
+
+import {Either, Option} from '../fp';
+
+import {ProtocolAction} from './actions';
+
+/*
+The ChannelState type is the data that protocols need about a given channel to decide what to do next.
+*/
+export type ChannelState = {
```

Comment:
Would this data be computed from the database? Or stored in the database?

Reply:
Computed from the database. So, the protocols do not need to care how states are stored